### PR TITLE
Setup codebase

### PR DESCRIPTION
**Pull request by Shinyaigeek**
## Detail

setup basic codebase for rust alternative aria-query

I made [aria_query workspace with rust](88d3f067265a914aeb8eba48675c4c934aa9f0a2), [license](da0e489d806e7ae6d026be5c83762714e9a8bfd5), [README](b38746af97af65d2c8a57b005a8324d7fbb1d4d8), [GHA workflow to check rust's codebase](7f7fcf451ce17bd70446e0fa77171dc48e55cee4) as I refered https://github.com/oxc-project/oxc

---

By the way, What should I deal with this aria-qeury's code review? I have some plans.

- I will make a pull request and merge it into `main` without review, and only if I want a review from the team I will send a review request
  - for example, when I need some discussion about library's I/O, license and code architecture
- I will make a pull request and merge it into `develop` branch without review, and I will make a pull request from `develop` to `main` when I finish implementing aria-query
- I will make a pull request and send a review request each time

Perhaps, if there is someone compatible for reviewer, please tell me 🙇

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2023-present Boshen
```
Comment (Shinyaigeek) [inline LICENSE:3]: https://github.com/oxc-project

> [@boshen](https://github.com/boshen) is the MIT license copyright holder, with publishing rights to [npmjs.com](https://npmjs.com/), [crates.io](https://crates.io/) and [marketplace.visualstudio.com/vscode](https://marketplace.visualstudio.com/vscode).

This is in response to this description of the oxc project

File: .cargo/config.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[alias]
```
Comment (Shinyaigeek) [inline .cargo/config.toml:1]: copied from https://github.com/oxc-project/oxc/blob/main/.cargo/config.toml

File: .github/actions/setup_rust_toolchain/action.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+name: Rustup
+
```
Comment (Shinyaigeek) [inline .github/actions/setup_rust_toolchain/action.yml:2]: copied from https://github.com/oxc-project/oxc/blob/main/.github/actions/rustup/action.yml

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,186 @@
+name: CI
```
Comment (Shinyaigeek) [inline .github/workflows/ci.yml:1]: copied from https://github.com/oxc-project/oxc/blob/main/.github/workflows/ci.yml

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+/target
```
Comment (Shinyaigeek) [inline .gitignore:1]: this is just a output of `cargo init` command, and I did not change any more

File: crates/aria_query/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,3 @@
+fn main() {
```
Comment (Shinyaigeek) [inline crates/aria_query/src/main.rs:1]: this is just a output of `cargo init` command, and I did not change any more

File: deny.toml [hunk 7]
```diff
@@ -0,0 +1,273 @@
+# This template contains all of the possible sections and their default values
```
Comment (Shinyaigeek) [inline deny.toml:1]: copied from https://github.com/oxc-project/oxc/blob/main/deny.toml

File: rust-toolchain.toml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+[toolchain]
```
Comment (Shinyaigeek) [inline rust-toolchain.toml:1]: copied from https://github.com/oxc-project/oxc/blob/main/rust-toolchain.toml

File: crates/aria_query/Cargo.toml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+[package]
```
Comment (Shinyaigeek) [inline crates/aria_query/Cargo.toml:1]: this is just a output of `cargo init` command, and I did not change any more

File: Cargo.toml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+[workspace]
```
Comment (Shinyaigeek) [inline Cargo.toml:1]: I copied some configuration needed from https://github.com/oxc-project/oxc/blob/main/Cargo.toml

**Comment by Boshen** [comment]
> I will make a pull request and merge it into main without review, and only if I want a review from the team I will send a review request
for example, when I need some discussion about library's I/O, license and code architecture

Let's start with this work flow, let's iterate fast and ship fast.

Rules and policies are documented here https://oxc-project.github.io/docs/contribute/rules.html for reference.

**Comment by Shinyaigeek** [comment]
ok sure, thank you!

**Pull request merged**